src/OpenSpaceToolkit/Astrodynamics/Trajectory/StateBuilder.cpp

```diff
@@ -74,6 +122,107 @@ const State StateBuilder::buildState(const Instant& anInstant, const VectorXd& a
     return {anInstant, aCoordinates, this->frameSPtr_, this->coordinatesBrokerSPtr_};
 }
 
+const State StateBuilder::reduce(const State& aState) const
+{
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("StateBuilder");
+    }
+
+    if (!aState.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("State");
+    }
+
+    if (aState.accessFrame() != this->frameSPtr_)
+    {
+        throw ostk::core::error::runtime::Wrong("State Frame");
+    }
+
+    VectorXd coordinates = VectorXd(this->coordinatesBrokerSPtr_->getNumberOfCoordinates());
+    Integer nextIndex = 0;
+
+    for (const auto& subset : this->coordinatesBrokerSPtr_->getSubsets())
+    {
+        if (!aState.accessCoordinatesBroker()->hasSubset(subset))
+        {
+            throw ostk::core::error::RuntimeError("Missing CoordinatesSubset: [{}]", subset->getName());
+        }
+
+        const VectorXd subsetCoordinates = aState.extractCoordinates(subset);
+        coordinates.segment(nextIndex, subsetCoordinates.size()) = subsetCoordinates;
+        nextIndex += subsetCoordinates.size();
+    }
```

Comment:
```suggestion
    VectorXd coordinates = aState.extractCoordinates(this->coordinatesBrokerSPtr_->getSubsets());
```

Since the State class has capability to extract multiple subsets at once, isn't it possible for this to be a one-liner?